build_tools/benchmarks/benchmark_helper.py

```diff
@@ -141,12 +145,76 @@ def _dump_cmds_handler(e2e_test_artifacts_dir: pathlib.Path,
   print(*lines, sep="\n")
 
 
+# Represents a benchmark results file with the data already loaded from a JSON file.
+class JSONBackedBenchmarkData:
+
+  def __init__(self, source_filepath: pathlib.PurePath, data: Dict):
+    if not isinstance(data, dict):
+      raise ValueError(
+          f"'{source_filepath}' seems not to be a valid benchmark-results-file (No JSON struct as root element)."
+      )
+    if "commit" not in data:
+      raise ValueError(
+          f"'{source_filepath}' seems not to be a valid benchmark-results-file ('commit' field not found)."
+      )
+    if "benchmarks" not in data:
+      raise ValueError(
+          f"'{source_filepath}' seems not to be a valid benchmark-results-file ('benchmarks' field not found)."
+      )
+
+    self.source_filepath: pathlib.PurePath = source_filepath
+    self.data: Dict = data
+
+  # Parses a JSON benchmark results file and makes some sanity checks
+  @staticmethod
+  def loadFromFile(filepath: pathlib.Path):
```

Comment:
Nit: We usually follow the naming convention `load_from_file`. Same in other places